test/micro/org/openjdk/bench/jdk/incubator/vector/ArrayMismatchBenchmark.java

```diff
@@ -0,0 +1,213 @@
+package org.openjdk.bench.jdk.incubator.vector;
+
+
+import jdk.incubator.vector.ByteVector;
+import jdk.incubator.vector.FloatVector;
+import jdk.incubator.vector.IntVector;
+import jdk.incubator.vector.LongVector;
+import jdk.incubator.vector.VectorMask;
+import jdk.incubator.vector.VectorOperators;
+import jdk.incubator.vector.VectorSpecies;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.infra.BenchmarkParams;
+
+import java.util.Arrays;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Thread)
+@Fork(jvmArgsPrepend = {"--add-modules=jdk.incubator.vector"})
+public class ArrayMismatchBenchmark {
+
+    @Param({"9", "257", "100000"})
+    int size;
+
+    @Param({"0.5", "1.0"})
+    double prefix;
+
+    byte[] byteData1;
+    byte[] byteData2;
+
+    int[] intData1;
+    int[] intData2;
+
+    long[] longData1;
+    long[] longData2;
+
+    float[] floatData1;
+    float[] floatData2;
+
+    static final VectorSpecies<Byte> BYTE_SPECIES_PREFERRED = ByteVector.SPECIES_PREFERRED;
+    static final VectorSpecies<Integer> INT_SPECIES_PREFERRED = IntVector.SPECIES_PREFERRED;
+    static final VectorSpecies<Float> FLOAT_SPECIES_PREFERRED = FloatVector.SPECIES_PREFERRED;
+    static final VectorSpecies<Long> LONG_SPECIES_PREFERRED = LongVector.SPECIES_PREFERRED;
+
+    static final Random random = new Random();
```

Comment:
We could use the recently added `RandomGenerator` instead in the spirit of encouraging the use of new and preferred APIs:

- remove the static field
- replace `FLOAT_SPECIES` with `DOUBLE_SPECIES`
- in `setup` create an instance `RandomGenerator rg = RandomGenerator.getDefault()`
- remove `createRandomFloats` and use `rg.doubles(...).toArray()`